File: src/dataset/cpp/beer-song_exercism-felbit/gemini/GeminiFlash_beer_song_v3.cpp
#include "beer_song.h"
#include <sstream>

std::string beer::verse(const short bottles) {
    std::stringstream ss;
    switch (bottles) {
        case 0:
            ss << "No more bottles of beer on the wall, no more bottles of beer.\n"
               << "Go to the store and buy some more, 99 bottles of beer on the wall.\n";
            break;
        case 1:
            ss << "1 bottle of beer on the wall, 1 bottle of beer.\n"
               << "Take it down and pass it around, no more bottles of beer on the wall.\n";
            break;
        case 2:
            ss << "2 bottles of beer on the wall, 2 bottles of beer.\n"
               << "Take one down and pass it around, 1 bottle of beer on the wall.\n";
            break;
        default:
            ss << bottles << " bottles of beer on the wall, " << bottles << " bottles of beer.\n"
               << "Take one down and pass it around, " << (bottles - 1) << " bottles of beer on the wall.\n";
            break;
    }
    return ss.str();
}

std::string beer::sing(const short start, const short end) {
    std::stringstream song;
    for (short it = start; it >= end; --it) {
        song << verse(it);
        if (it != end) {
            song << "\n";
        }
    }
    return song.str();
}

File: src/dataset/Java/beer-song_exercism-uzilan/gemini/GeminiFlash_BeerSong_v3.java
import static java.lang.Character.toUpperCase;

public class BeerSong {

    public String sing(final int from, final int count) {
        StringBuilder song = new StringBuilder();
        int start = Math.max(0, from - count + 1);

        for (int i = from; i >= start; i--) {
            if (i == 0) {
                song.append(capitalize(bottles(i))).append(" of beer on the wall, ").append(bottles(i)).append(" of beer.\n");
                song.append("Go to the store and buy some more, ").append(bottles(99)).append(" of beer on the wall.\n");
            } else {
                song.append(bottles(i)).append(" of beer on the wall, ").append(bottles(i)).append(" of beer.\n");
                song.append("Take ").append(one(i)).append(" down and pass it around, ").append(bottles(i - 1)).append(" of beer on the wall.\n");
            }
            if (i > start) {
                song.append("\n");
            }
        }
        return song.toString();
    }

    public String singSong() {
        return sing(99, 100);
    }

    private String bottles(int count) {
        switch (count) {
            case 0:
                return "no more bottles";
            case 1:
                return "1 bottle";
            default:
                return count + " bottles";
        }
    }

    private String capitalize(String string) {
        if (string == null || string.isEmpty()) {
            return string;
        }
        return toUpperCase(string.charAt(0)) + string.substring(1);
    }

    private String one(int count) {
        return (count == 1) ? "it" : "one";
    }
}